crates/wast/src/wast.rs

```diff
@@ -535,3 +555,14 @@ fn is_arithmetic_f32_nan(bits: &u32) -> bool {
 fn is_arithmetic_f64_nan(bits: &u64) -> bool {
     return (bits & 0x0008000000000000) == 0x0008000000000000;
 }
+
+fn values_equal(v1: &Val, v2: &Val) -> bool {
+    match (v1, v2) {
+        (Val::I32(a), Val::I32(b)) => a == b,
+        (Val::I64(a), Val::I64(b)) => a == b,
+        (Val::F32(a), Val::F32(b)) => a == b,
+        (Val::F64(a), Val::F64(b)) => a == b,
```

Comment:
While here, it's worth a comment mentioning that f32/f64 cases here are actually doing integer comparisons, so they distinguish between 0 and +0 as expected.